pkg/orchestrator/etcd_worker.go

```diff
@@ -362,12 +366,15 @@ func (worker *EtcdWorker) commitChangedState(ctx context.Context, changedState m
 
 	worker.metrics.metricEtcdTxnSize.Observe(float64(size))
 	startTime := time.Now()
-	resp, err := worker.client.Txn(ctx).If(cmps...).Then(ops...).Commit()
-	costTime := time.Since(startTime).Seconds()
-	if costTime > time.Second.Seconds()*1 {
-		log.Warn("etcdWorker commit etcd txn cost time more than 1 second")
+
+	ctx1, cancel := context.WithTimeout(ctx, etcdTxnTimeoutDuration)
+	defer cancel()
```

Comment:
Probably we should cancel the `ctx1` immediately after the txn is committed. Refer to https://pkg.go.dev/context#WithTimeout

Reply:
ok